src/rdwin32.h

```diff
@@ -377,6 +377,32 @@ static RD_UNUSED int rd_pipe_nonblocking(rd_socket_t *fds) {
 #define rd_open(path, flags, mode) _open(path, flags, mode)
 #define rd_close(fd)               _close(fd)
 
+/**
+ * PRNG
+ */
+
+/**
+ * @brief rand_r() for Windows systems to ensure thread agnostic rand() results
+ * @returns a pseudo-random integer in the range 0 to RAND_MAX inclusive [0,
+ * RAND_MAX].
+ */
+static RD_UNUSED RD_INLINE int
+rd_rand_r(unsigned int *seed) {
+        /* Source: https://git.musl-libc.org/cgit/musl/tree/src/prng/rand_r.c */
```

Comment:
This has license implications.

Since your implementation here is not a verbatim copy of musl's implementation, I believe it is safe to say that this implementation is inspired by the above link, rather than calling it source.

Alternatively we add a rdrandr.c that is only built if HAVE_RAND_R is not set and then add a LICENSE.rdrandr to the top-level directoy containing which file is covered (src/rdrandr.c) and the musl license (MIT).